STEP/TypeVisitor.cs

```diff
@@ -194,26 +194,32 @@ public class TypeVisitor : IVisitor {
         DclVisitor dclVisitor = new DclVisitor(_symbolTable);
         n.Left.Accept(dclVisitor);
         n.Right.Accept(this);
-        if (n.Left.Type.ActualType is TypeVal.Analogpin or TypeVal.Digitalpin) { // Should this be a separate method or PinVisitor?
-            int pinVal = (int) ((NumberNode) n.Right).Value;
-            switch (n.Left.Type.ActualType) {
-                case TypeVal.Analogpin when pinVal is < 0 or > 5:
-                    throw new ArgumentOutOfRangeException(nameof(pinVal), "Analog pins must be in range 0-5");
-                case TypeVal.Digitalpin when pinVal is < 0 or > 13:
-                    throw new ArgumentOutOfRangeException(nameof(pinVal), "Digital pins must be in range 0-13");
-                default:
-                    _pinTable.RegisterPin(n.Left.Type.ActualType, pinVal);
-                    break;
-            }
+        if (n.Left.Type == n.Right.Type) {
```

Comment:
Bør den eksplicit tjekke om de er andet end analogpin og digitalpin?

Reply:
Det burde være nok at tjekke de to objects for equality, right? Hvis de er andet end analogpin eller digitalpin, så bør de ikke være equal.